client/verta/verta/_deployment/endpoint.py

```diff
@@ -73,18 +71,27 @@ def _get_or_create_by_name(cls, conn, name, getter, creator):
     def _get_by_path(cls, conn, conf, workspace, path):
         endpoint_json = cls._get_json_by_path(conn, workspace, path)
         if endpoint_json:
-            return cls(conn, conf, workspace, endpoint_json['id'])
+            return Endpoint._new(conf, conn, workspace, endpoint_json)
         else:
             return None
 
+    @classmethod
+    def _new(cls, conf, conn, workspace, endpoint_json):
+        return cls(conn, conf, workspace, endpoint_json['id'], endpoint_json['creator_request']['path'])
```

Comment:
I do not think it is a good design pattern to use a `obj._new()` instead of `obj.__init__()`. Also see my comment on `__init__()`
```suggestion
```

Reply:
changed to constructor